pkg/shared/pubsub/testpubsub/pubsub_implementation_test.go

```diff
@@ -0,0 +1,33 @@
+package testpubsub
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/mint-app/pkg/shared/pubsub"
+	"github.com/cmsgov/mint-app/pkg/shared/pubsub/mockpubsub"
+)
+
+func TestPubSubImplementation_Publish(t *testing.T) {
+	ps, subscriber, eventType, payload, modelPlanID, _, disconnectChannel := setupPubSubTest(t)
+
+	subscriber.EXPECT().GetID().Return("MOCK_SUBSCRIBER").Times(2)
+	subscriber.EXPECT().Notify(payload)
+
+	ps.Subscribe(modelPlanID, eventType, subscriber, disconnectChannel)
+	ps.Publish(modelPlanID, eventType, payload)
+}
+
+func setupPubSubTest(t *testing.T) (*pubsub.ServicePubSub, *mockpubsub.MockSubscriber, pubsub.EventType, string, uuid.UUID, string, chan struct{}) {
```

Comment:
For what it's worth (and I know there's been some discussion around `setupPubSubTest()` here already), I've taken an approach of using Testify to setup tests (and using their [SetupTest()](https://pkg.go.dev/github.com/stretchr/testify/suite?utm_source=godoc#SetupTestSuite) function - [example in MINT](https://github.com/CMSgov/mint-app/blob/3eb0efae59847b8776b413b42fe9945f84b60622/pkg/graph/resolvers/resolver_test.go#L20-L23))

This lets you tack on items to the `suite` object itself, making them globally available to each test, which is especially nice if you have test-specific dependencies (like a pubsub instance, subscriber instance, etc.) that you want populated for each test.

Might be overkill here, since it's just the 1 test using it, but I think it could be nice if we started doing this across our tests!

Reply:
Took a stab, will implement in a second pass